### PR TITLE
Update boost to v1.81.0

### PR DESCRIPTION
**Pull request by tnixeu**
Adds new boost lib Boost::url

* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/update.html)
  step by step carefully. **Yes**

### CODE REVIEW

**Comment by NeroBurner** [comment]
I think the `matrix.json` override somewhere has a `,` too much or is missing one somewhere

from the `set_matrix` logs:
```
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 152 column 1 (char 21604)
```

**Comment by tnixeu** [comment]
One was missing. Copy and waste :/

File: cmake/projects/Boost/ci/matrix.json [hunk 1]
```diff
@@ -233,6 +233,21 @@
...
+// { "example": "Boost-url", "toolchain": "osx-12-3-arch-universal2-cxx17",                  "os": "macos-12",     "python": "3.8", "script": "build.sh"  },
+{ "example": "Boost-url", "toolchain": "ios-nocodesign-15-5-arm64-cxx17",                 "os": "macos-12",     "python": "3.8", "script": "build.sh"  },
+// { "example": "Boost-url", "toolchain": "ninja-vs-16-2019-win64-cxx17",                    "os": "windows-2019", "python": "3.8", "script": "build.cmd" },
+{ "example": "Boost-url", "toolchain": "nmake-vs-16-2019-win64-cxx17",                    "os": "windows-2019", "python": "3.8", "script": "build.cmd" },
+// { "example": "Boost-url", "toolchain": "vs-16-2019-win64-sdk-10-0-18362-0-cxx17",         "os": "windows-2019", "python": "3.8", "script": "build.cmd" },
+{ "example": "Boost-url", "toolchain": "mingw-cxx17",                                     "os": "windows-2019", "python": "3.8", "script": "build.cmd" },
+{ "example": "Boost-url", "toolchain": "msys-cxx17",                                      "os": "windows-2019", "python": "3.8", "script": "build.cmd" }
+
```
Comment (NeroBurner) [inline cmake/projects/Boost/ci/matrix.json:251]: one added extra line, don't need it, remove again
```suggestion
```

**Review by NeroBurner — APPROVED** [review]
looking good, thanks for the update. going to merge after the toolchains ran through

**Comment by bothefirst** [comment]
@NeroBurner when is the new release containing this PR is scheduled ?

**Comment by NeroBurner** [comment]
I think I'm gonna do one tomorrow. You're probably eager to use the new Boost::url package? :)

**Comment by bothefirst** [comment]
yes  + a bunch of exciting improvements on Asio :-) thanx !

**Comment by NeroBurner** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.24.12

**Pull request merged**